packages/core/src/components/GenericFeature/GenericFeature.tsx

```diff
@@ -0,0 +1,156 @@
+*     {
+*       name: 'JsonForm', // import from @mock-app/components
+*       type: 'EntryData' as ComponentType,
+*       props: {
+*         schema: {
+*           type: 'object',
+*           properties: {
+*             name: {
+*               type: 'string',
+*             },
+*             vegetarian: {
+*               type: 'boolean',
+*             },
+*           },
+*         },
+*         onChange: (value: any) => {
+*           setJsonFormData(value.data);
+*         },
+*         uiSchema: {
+*           type: 'VerticalLayout',
+*           elements: [
+*             {
+*               type: 'Control',
+*               scope: '#/properties/name',
+*            },
+*             {
+*               type: 'Control',
+*               scope: '#/properties/vegetarian',
+*             },
+*          ],
+*         },
+*         initialData: {},
+*         className: '',
+*       },
+*     },
+*     {
+*       name: 'Button',
+*       type: 'Submit' as ComponentType,
+*       props: {},
+*     },
+*  ];
+*
+*    const services = [
+*    {
+*      name: 'logService',
+*      parameters: [jsonFormData],
+*    },
+*    {
+*      name: 'logServiceTwo',
+*      parameters: [],
+*    },
+*  ];
+* return (
+*    <div className='example'>
+*      <GenericFeature components={componentsData} services={services} />
+*    </div>
+*    );
+*  }
+*/
+
+import React from 'react';
+import * as services from '@mock-app/services';
+// import * as events from '@mock-app/events';
+import { ComponentType, DynamicComponentRenderer, IDynamicComponentRendererProps } from './DynamicComponentRenderer';
+
+export interface IServiceDefinition {
+  name: string;
+  parameters: any[];
+}
+
+export interface IGenericFeatureProps {
+  components: IDynamicComponentRendererProps[];
+  services: IServiceDefinition[];
+}
+
+const getService = (name: string) => {
+  const serviceName = name as keyof typeof services;
+  if (typeof services[serviceName] !== 'undefined') {
+    return services[serviceName];
+  }
+  // const eventName = name as keyof typeof events;
+  // if (typeof events[eventName] !== 'undefined') {
+  //   return events[eventName];
+  // }
+  throw new Error(`Service or event ${name} not found`);
+};
+
+/**
+ *
+ * @param param0
+ * @returns
+ */
+export const GenericFeature: React.FC<IGenericFeatureProps> = ({ components, services }: IGenericFeatureProps) => {
+  const [state, setState] = React.useState<any[]>([]);
+  const props: Record<string, any> = {};
+
+  const executeServices = async (services: IServiceDefinition[], parameters: any[]) => {
+    return services.reduce(async (previousResult: any[] | Promise<any[]>, currentService) => {
+      const prevResult = await previousResult;
+      const service: any = getService(currentService.name);
+      const params = [...prevResult, ...currentService.parameters];
+      const result: any = service(...params);
+      return [result];
+    }, parameters);
+  };
```

Comment:
I don't see any function in this code block running asynchronously. Could you please explain why we use async/await here?

Reply:
This function executes the services with the accumulated parameters. We should put it asynchronously because it may have many services executed. For example, if we have two services, the second service should wait until the first service returns the value and receive this value as a param for the next service function.